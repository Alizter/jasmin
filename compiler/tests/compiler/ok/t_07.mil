fn foo0(z : stack u64) -> stack u64 {
  z += z;
  return z;
}

fn foo1(x, y : stack u64, cf : reg bool) -> stack u64 {
  cf,x += y + cf;
  if cf {
    cf, x += y;
    cf, x += y;
  }
  x = foo0(x);
  return x;
}

fn foo2(x, y : stack u64, cf : reg bool) -> stack u64 {
  w : stack u64;
  w = x;
  cf,w += x;
  if !cf {
    cf, x += w;
    cf, x += y;
    x = foo1(x,y,cf);
  }
  return x;
}

/*
START:CMD
ARG="inline[foo2],renumber_reuse,print[input][fun=foo2,info]"
END:CMD
*/
