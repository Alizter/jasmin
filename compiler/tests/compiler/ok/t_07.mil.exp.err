Processed file tests/compiler/ok/t_07.mil
