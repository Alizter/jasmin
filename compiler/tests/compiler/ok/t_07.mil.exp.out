inlining all calls in foo2 function(s)
   REMOVED_TIME
renumbering identifiers in all function(s)
   REMOVED_TIME
>> input:
fn foo2(x : stack u64, y : stack u64, cf : reg bool) -> stack u64 {
  cf.1 : reg bool
  w : stack u64
  x.1 : stack u64
  y.1 : stack u64
  z : stack u64
  // START
  w = x;
  cf, w += x;
  // END
  // START
  if !cf {
    // START
    cf, x += w;
    cf, x += y;
    /* START: inlined call to foo1 */
    x.1 := x;
    y.1 := y;
    cf.1 := cf;
    cf.1, x.1 += y.1 + cf.1;
    // END
    // START
    if cf.1 {
      // START
      cf.1, x.1 += y.1;
      cf.1, x.1 += y.1;
      // END
    }
    // END
    // START
    /* START: inlined call to foo0 */
    z := x.1;
    z += z;
    x.1 := z;
    /* END: inlined call to foo0 */
    x := x.1;
    /* END: inlined call to foo1 */
    // END
  }
  // END
  return x;
}

